### PR TITLE
Code changes for JsonObject Note element Not Mandatory

### PR DESCRIPTION
**Pull request by RameshRapa**
Added code to check for note element availability in the JsonObject. if not available setting it to null

### CODE REVIEW

File: src/org/labkey/snd/SNDController.java [hunk 1]
```diff
@@ -1006,7 +1006,7 @@ public ApiResponse execute(SimpleApiJsonForm form, BindException errors)
             if (!errors.hasErrors())
             {
                 String projectIdrev = json.getString("projectIdRev");
-                String note = json.getString("note");
+                String note = json.has("note") ? json.getString("note") : null;
```
Comment (spamhurts) [inline src/org/labkey/snd/SNDController.java:1009]: Do we need to also check for an empty string?

Comment (spamhurts) [reply src/org/labkey/snd/SNDController.java:1009]: json.has("note") && json.getString("note").length() > 0

Comment (RameshRapa) [reply src/org/labkey/snd/SNDController.java:1009]: the empty string is being checked in our Mobile application while adding the value and purging it if not needed and the note element will not be present in the Json object and when it is not present , i am setting null. i validated with postman and also ran application with the code modified above.

Comment (spamhurts) [reply src/org/labkey/snd/SNDController.java:1009]: Works for me - thanks!

**Review by spamhurts — APPROVED** [review]

**Pull request merged**